configure.in

```
@@ -261,6 +261,17 @@ if test "$HAS_TABLEAM" == 'yes'; then
 
 fi # test "$HAS_TABLEAM" == 'yes'
 
+
+PGAC_ARG_BOOL(with, security-flags, yes,
+              [do not use security flags])
+AC_SUBST(with_security_flags)
+
+if test "$with_security_flags" = yes; then
+SECURITY_FLAGS="-fstack-protector-strong -D_FORTIFY_SOURCE=2 -O2 -z noexecstack -fpic -Wl,-z,relro -Wl,-z,now -Wformat -Wformat-security -Werror=format-security"
+CITUS_CFLAGS="$CITUS_CFLAGS $SECURITY_FLAGS"
```

I think it would be good to add these flags to `CITUS_CPPFLAGS` too. These are used when generating JIT code. 

I do get some warnings about linker flags not being used when simply adding this line and configuring postgres with `--with-llvm` (to generate jit code):
```suggestion
CITUS_CPPFLAGS="$CITUS_CPPFLAGS $SECURITY_FLAGS"
```
So it might be best to split the linker and the compiler flags in two pieces and add the linker security flags to `CITUS_LDFLAGS`.

---

CPPFLAGS are used by both the gcc and clang and I can not put anything that does not work in both compilers in CPPFLAGS.

I ended up splitting everything into 3:
- place it in CFLAGS if it is a compiler flag potentially supported by gcc
- place it in LDFLAGS if it is a linker flag
- place it in BITCODE_CFLAGS if it is a clang flag.

Notes:
1) There are many CLANG flags that we have missed in the past. I added all of them to BITCODE_FLAGS
2) .bc files are not linked and hence no LDFLAG necessary for llvm